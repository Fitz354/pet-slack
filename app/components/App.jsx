import React from 'react';
import ChannelsList from '../components/ChannelsList.jsx';

export default () => (
  <div className="container">
    <div className="row">
      <div className="col-">
        <ChannelsList />
      </div>
      <div className="col-xl">chat</div>
    </div>
  </div>
);
