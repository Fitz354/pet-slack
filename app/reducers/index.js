import { handleActions } from 'redux-actions';
import { combineReducers } from 'redux';
import { reducer as formReducer } from 'redux-form';
import * as actions from '../actions';

const messageCreatingState = handleActions({
  [actions.sendMessageRequest]() {
    return 'requested';
  },
  [actions.sendMessageSuccess]() {
    return 'successed';
  },
  [actions.sendMessageFailure]() {
    return 'failed';
  },
}, 'none');

const messages = handleActions({
  [actions.getMessage](state, { payload: { message } }) {
    return [...state, message];
  },
  [actions.sendMessageSuccess](state, { payload: { message } }) {
    return [...state, message];
  },
}, []);

const channels = handleActions({}, []);
const currentChannelId = handleActions({
  [actions.switchChannel](state, { payload: { id } }) {
    return id;
  },
}, 0);

export default combineReducers({
  messageCreatingState,
  channels,
  messages,
  currentChannelId,
  form: formReducer,
});
