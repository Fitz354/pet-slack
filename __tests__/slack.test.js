import 'js-polyfills/html';

import React from 'react';
import { Provider } from 'react-redux';
import { createStore, applyMiddleware } from 'redux';
import thunk from 'redux-thunk';

import Enzyme, { mount } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import timeout from 'timeout-then';

import axios from 'axios';
import nock from 'nock';
import httpAdapter from 'axios/lib/adapters/http';

import App from '../app/components/App.jsx';
import reducers from '../app/reducers';

Enzyme.configure({ adapter: new Adapter() });

axios.defaults.adapter = httpAdapter;
nock.disableNetConnect();

const host = 'http://localhost';

it('check app', async () => {
  document.body.innerHTML = '<div id="app-container"></div>';
  const initState = {
    userName: 'user',
    channels: [
      { id: 1, name: 'general', removable: false },
      { id: 2, name: 'random', removable: false },
    ],
    messages: [],
    currentChannelId: 1,
  };
  const store = createStore(
    reducers,
    initState,
    applyMiddleware(thunk),
  );

  const wrapper = mount(
    <Provider store={store}>
      <App />
    </Provider>,
    document.getElementById('app-container'),
  );

  expect(wrapper.render()).toMatchSnapshot();

  const messageInput = wrapper.find('.message-input input[type="text"]');
  const messageSubmit = wrapper.find('.message-submit');

  messageInput.simulate('change', { target: { value: 'hey!' } });
  expect(wrapper.render()).toMatchSnapshot();

  nock(host)
    .post('/api/v1/channels/1/messages')
    .reply(200, (uri, body) => {
      const data = JSON.parse(body);
      const attributes = {
        ...data.data.attributes,
        time: '10:00',
        id: 1,
      };

      const response = {
        data: {
          type: 'messages',
          attributes,
        },
      };

      return response;
    });

  messageSubmit.simulate('submit');
  await timeout(1000);
  expect(wrapper.render()).toMatchSnapshot();
});
