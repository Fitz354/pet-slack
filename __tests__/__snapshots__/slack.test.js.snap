// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`check app 1`] = `
<div
  class="wrapper container-fluid h-100"
>
  <div
    class="row h-100"
  >
    <div
      class="w-25 h-100 pt-3 bg-dark"
    >
      <div
        class="channels-list h-100"
      >
        <div
          class="user-name pl-3 pr-3"
        >
          <p
            class="h3 text-white"
          >
            user
          </p>
        </div>
        <div
          class="nav flex-column nav-pills"
        >
          <a
            class="nav-link text-white active"
            href="#"
          >
            # general
          </a>
          <a
            class="nav-link text-white"
            href="#"
          >
            # random
          </a>
        </div>
      </div>
    </div>
    <div
      class="col pt-3 pb-5 h-100 bg-light"
    >
      <div
        class="chat d-flex flex-column justify-content-between h-100"
      >
        <h2
          class="chat-name"
        >
          #general
        </h2>
        <div
          class="chat-messages col"
        >
          <div
            class="card h-100"
          >
            <div
              class="card-body"
            />
          </div>
        </div>
        <div
          class="chat-input mt-3"
        >
          <form
            class="form-inline"
          >
            <div
              class="form-group w-50 mr-3"
            >
              <input
                class="message-input form-control w-100"
                name="message"
                placeholder="Message to #general"
                type="text"
                value=""
              />
            </div>
            <button
              class="message-submit btn btn-primary"
              type="submit"
            >
              Send
            </button>
          </form>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`check app 2`] = `
<div
  class="wrapper container-fluid h-100"
>
  <div
    class="row h-100"
  >
    <div
      class="w-25 h-100 pt-3 bg-dark"
    >
      <div
        class="channels-list h-100"
      >
        <div
          class="user-name pl-3 pr-3"
        >
          <p
            class="h3 text-white"
          >
            user
          </p>
        </div>
        <div
          class="nav flex-column nav-pills"
        >
          <a
            class="nav-link text-white active"
            href="#"
          >
            # general
          </a>
          <a
            class="nav-link text-white"
            href="#"
          >
            # random
          </a>
        </div>
      </div>
    </div>
    <div
      class="col pt-3 pb-5 h-100 bg-light"
    >
      <div
        class="chat d-flex flex-column justify-content-between h-100"
      >
        <h2
          class="chat-name"
        >
          #general
        </h2>
        <div
          class="chat-messages col"
        >
          <div
            class="card h-100"
          >
            <div
              class="card-body"
            />
          </div>
        </div>
        <div
          class="chat-input mt-3"
        >
          <form
            class="form-inline"
          >
            <div
              class="form-group w-50 mr-3"
            >
              <input
                class="message-input form-control w-100"
                name="message"
                placeholder="Message to #general"
                type="text"
                value="hey!"
              />
            </div>
            <button
              class="message-submit btn btn-primary"
              type="submit"
            >
              Send
            </button>
          </form>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`check app 3`] = `
<div
  class="wrapper container-fluid h-100"
>
  <div
    class="row h-100"
  >
    <div
      class="w-25 h-100 pt-3 bg-dark"
    >
      <div
        class="channels-list h-100"
      >
        <div
          class="user-name pl-3 pr-3"
        >
          <p
            class="h3 text-white"
          >
            user
          </p>
        </div>
        <div
          class="nav flex-column nav-pills"
        >
          <a
            class="nav-link text-white active"
            href="#"
          >
            # general
          </a>
          <a
            class="nav-link text-white"
            href="#"
          >
            # random
          </a>
        </div>
      </div>
    </div>
    <div
      class="col pt-3 pb-5 h-100 bg-light"
    >
      <div
        class="chat d-flex flex-column justify-content-between h-100"
      >
        <h2
          class="chat-name"
        >
          #general
        </h2>
        <div
          class="chat-messages col"
        >
          <div
            class="card h-100"
          >
            <div
              class="card-body"
            >
              <div
                class="message"
              >
                <div
                  class="message-header"
                >
                  <span
                    class="message-sender font-weight-bold"
                  >
                    user 
                  </span>
                  <span
                    class="message-time"
                  >
                    10:00
                  </span>
                </div>
                <p
                  class="pl-3 pr-3"
                >
                  hey!
                </p>
              </div>
            </div>
          </div>
        </div>
        <div
          class="chat-input mt-3"
        >
          <form
            class="form-inline"
          >
            <div
              class="form-group w-50 mr-3"
            >
              <input
                class="message-input form-control w-100"
                name="message"
                placeholder="Message to #general"
                type="text"
                value=""
              />
            </div>
            <button
              class="message-submit btn btn-primary"
              type="submit"
            >
              Send
            </button>
          </form>
        </div>
      </div>
    </div>
  </div>
</div>
`;
